.github/workflows/terraform-test.yml

```diff
@@ -0,0 +1,28 @@
+name: 'Terraform Module Test'
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  tf-test:
+    runs-on: ubuntu-latest
+    permissions: {}
+    steps:
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332
+      - uses: hashicorp/setup-terraform@651471c36a6092792c552e8b1bef71e592b462d8
+        with:
+          terraform_version: 1.9.2
```

Comment:
you can use 1.9.x better. 

still better to create a GitHub variable like in the TF repo, e.g. TF_VERSION_LATEST